tasks_test.go

```diff
@@ -0,0 +1,180 @@
+package druid
+
+import (
+	"bytes"
+	"context"
+	"errors"
+	"testing"
+	"time"
+
+	"github.com/gocarina/gocsv"
+	"github.com/google/uuid"
+	"github.com/jmoiron/sqlx/types"
+	"github.com/stretchr/testify/require"
+	tc "github.com/testcontainers/testcontainers-go/modules/compose"
+	"github.com/testcontainers/testcontainers-go/wait"
+)
+
+// TestDO represents entry with payload.
+type TestDO struct {
+	Timestamp time.Time      `db:"ts"`
+	Id        uuid.UUID      `db:"id"`
+	Payload   types.JSONText `db:"payload"`
+}
+
+var testObjects = []TestDO{
+	{
+		Id:        uuid.New(),
+		Timestamp: time.Now(),
+		Payload:   types.JSONText("{\"test\": \"json\"}"),
+	},
+	{
+		Id:        uuid.New(),
+		Timestamp: time.Now().Add(time.Hour),
+		Payload:   types.JSONText("{\"test\": \"json2\"}"),
+	},
+}
+
+func triggerIngestionTask(d *Client, dataSourceName string, entries []TestDO) (string, error) {
+	csvEntriesBuff := &bytes.Buffer{}
+
+	err := gocsv.MarshalWithoutHeaders(entries, csvEntriesBuff)
+	if err != nil {
+		return "", err
+	}
+
+	var spec = NewTaskIngestionSpec(
+		SetTaskType("index_parallel"),
+		SetTaskDataSource(dataSourceName),
+		SetTaskTuningConfig("index_parallel", 25000, 5000000),
+		SetTaskIOConfigType("index_parallel"),
+		SetTaskInputFormat("csv", "false", []string{"ts", "id", "payload"}),
+		SetTaskInlineInputData(csvEntriesBuff.String()),
+	)
+	taskID, err := d.Tasks().SubmitTask(spec)
+	return taskID, err
+}
+
+func awaitTaskCompletion(client *Client, taskID string) error {
```

Comment:
These functions do not have timeout. If things go wrong running such test can potentially consume all CI minutes. Please add some sane long timeout.

Reply:
Thanks. Fixed.